### DISCUSSION TITLE
Why can I not Request Changes as the Author?

### DISCUSSION THREAD

**Discussion by mve-oncomfort** [community/community#39236] [Pull Requests] [answered]
### Select Topic Area

Product Feedback

### Body

Why can I not request changes as the author of a pull request?
![image](https://user-images.githubusercontent.com/97109949/202142654-a873da03-7d16-4605-8a5b-20235fa2ea4e.png)

Then how do I create a pull request, which is able to receive comments, without it being able to be merged accidentally?

**Comment by Preetiraj3697** [comment]
You cannot change the author of a pull request or create one in somebody else's name. GitHub doesn't allow this because it would permit people to forge pull requests and let another person receive negative feedback. For example, if I opened a PR in your name containing something wildly inappropriate (say, a racist slur), you'd be blamed for it unfairly. Moreover, in some regulated environments, keeping an accurate record of who contributed code is very important for legal reasons, which is another reason why this isn't permitted.

In most cases, if a commit gets reverted, then just have the original author open a new PR. Clearly they'd wanted to submit the code in the first place. You can even create the branch for them with their commit squashed and prepared, as long as they open the PR. This is a case where a helpful conversation with another contributor goes a long way, and presumably you'd want to let them know in a kind way that they made a mistake and that you reverted it anyway. Since there's no way to create a repository on behalf of another user, you should close your PR and ask them original submitter to open it instead.

I think it's obvious why you can't approve your own PRs. I can see that this is a situation where requesting changes might be useful, but I suppose the rationale is that you wouldn't have opened the PR if you thought it needed improvements, and if you later learned that, you certainly would revise it instead of merging it.

**Comment by airtower-luna** [comment, marked as answer]
It's your PR, which means if you want changes you can just make them, so it wouldn't make sense for you to request changes. If you want to say that your PR isn't ready yet, [mark it as a draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request#converting-a-pull-request-to-a-draft) (you can also do that while creating the PR).

**Reply by oakkitten** [reply]
Multiple people can push to a PR, and if someone other than the original author is currently working on it, it makes sense to allow the original author to request changes from them.

**Reply by mve-oncomfort** [reply]
Thank you, that's what I am looking for. Sometimes it is useful to already create the PR and collect comments even when we know some things still need to be changed. For me, the Pull Request is a collaboration tool.
I still think authors should be able to request changes. I don't see why this feature is artificially blocked. At least make it a setting in the repo settings.

I see I don't have access to the draft feature since I am using a private repository on a free account.
> Draft pull requests are available in public repositories with GitHub Free for organizations and legacy per-repository billing plans, and in public and private repositories with GitHub Team, GitHub Enterprise Server 2.17+, and GitHub Enterprise Cloud. For more information, see "[GitHub's products](https://docs.github.com/en/get-started/learning-about-github/githubs-products)."
